tests/attr/test_saliency.py

```diff
@@ -136,6 +136,13 @@ def test_saliency_classification_smoothgrad(self) -> None:
     def test_saliency_classification_vargrad(self) -> None:
         self._saliency_classification_assert(nt_type="vargrad")
 
+    def test_saliency_grad_unchanged(self) -> None:
+        model, inp, grads, add_args = _get_basic_config()
+        inp.grad = torch.randn_like(inp)
+        grad = inp.grad.detach().clone()
+        self._saliency_base_assert(model, inp, grads, add_args)
+        assertTensorTuplesAlmostEqual(self, inp.grad, grad)
```

Comment:
nit: Can we do exact equal here to be sure that the grads didn't change ?
It would be interesting to use backward and compare with the grads computed with the backward to contrast the difference.

Reply:
Sure, updated with delta=0.0 for exact equality. From the docs, I think backward should always compute the same gradients just also updating the grad attribute for leaf nodes, but can double-check this.

Reply:
Thank you! I was thinking that if we use backward instead autograd.grad then we will see the difference, right ?

Reply:
Makes sense, yup, if we used backward instead, this test should fail! Although backward doesn't take inputs as parameters and just requires accessing grad for the gradients, so would be a different structure from what we currently use and might fail for non-leaf inputs.